cyral/resource_cyral_sidecar_listener_test.go

```diff
@@ -0,0 +1,124 @@
+package cyral
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	sidecarListenerResourceName    = "cyral_sidecar_listener"
+	plainMySQLListenerResourceName = "plain_mysql"
+	plainMySQLListener             = `resource "%s" "%s" {
+  sidecar_id = %s
+  tcp_listener {
+    port = 3306
+  }
+  repo_types =["mysql"]
+}
+`
+	multiplexedMySQLListenerResourceName = "multiplexed_mysql"
+	multiplexedMySQLListener             = `resource "%s" "%s" {
+    sidecar_id = %s
+    tcp_listener {
+		port = 3307
+    }
+    multiplexed = true
+    mysql_settings {
+		db_version = "5.7"
+    }
+    repo_types =["mysql"]
+}
+`
+	s3ListenerResourceName = "s3"
+	s3Listener             = `resource "%s" "%s" {
+    sidecar_id = %s
+    tcp_listener {
+		port = 443
+	}
+    s3_settings {
+		proxy_mode = true
+    }
+    repo_types =["s3"]
+}
+`
+	mariaDbSocketListenerResourceName = "maria_db_socket"
+	mariaDbSocketListener             = `resource "%s" "%s" {
+    sidecar_id = %s
+    unix_listener {
+		file = "/var/run/mysqld/mysql.sock"
+    }
+    repo_types =["mariadb"]
+}
+`
+)
+
+func TestSidecarListenerResource(t *testing.T) {
+
+	testConfig, testFunc := setupSidecarListenerTest()
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testConfig,
```

Comment:
I would recommend creating different steps for each listener configuration instead of testing everything together in a single step. This is just a suggestion though, feel free to ignore it.